src/Chip/Chip.spec.js

```diff
@@ -0,0 +1,76 @@
+// @flow weak
+/* eslint-env mocha */
+
+import React from 'react';
+import { assert } from 'chai';
+import { createShallowWithContext } from 'test/utils';
+import Chip, { styleSheet } from './Chip.js';
+import Avatar from '../Avatar/Avatar.js';
+
+describe('<Chip />', () => {
+  let shallow;
+  let classes;
+
+  before(() => {
+    shallow = createShallowWithContext();
+    classes = shallow.context.styleManager.render(styleSheet);
+  });
+
+  describe('Text Chip', () => {
+    let wrapper;
+
+    before(() => {
+      wrapper = shallow(
+        <Chip
+          className="my-Chip"
+          data-my-prop="woof"
+        >
+          Text Chip
+        </Chip>,
+      );
+    });
+
+    it('should render a div containing a span', () => {
+      assert.strictEqual(wrapper.is('div'), true, 'should be a div');
+      assert.strictEqual(wrapper.childAt(0).is('span'), true, 'should be a span');
+    });
+
+    it('should spread custom props merge user classes to the root node', () => {
+      assert.strictEqual(wrapper.hasClass(classes.root), true);
+      assert.strictEqual(wrapper.hasClass('my-Chip'), true);
+      assert.strictEqual(wrapper.prop('data-my-prop'), 'woof');
+    });
+  });
+
+  describe('Deletable Avatar Chip', () => {
+    let wrapper;
+
+    before(() => {
+      wrapper = shallow(
+        <Chip
+          onClick={() => {}}
+          onRequestDelete={() => {}}
+          className="my-Chip"
+          data-my-prop="woof"
+        >
+          <Avatar>MB</Avatar>
+          Text Avatar Chip
+        </Chip>,
+      );
+    });
+
+    it('should render a div containing an Avatar div, span and svg', () => {
+      assert.strictEqual(wrapper.is('div'), true, 'should be a div');
+      assert.strictEqual(wrapper.childAt(0).prop('component'), 'div', 'should be a div');
```

Comment:
I was struggling to find the right target for the test, so arrived at that syntax blindly, so more than happy to change it if this is the right thing to test for...